src/Microsoft.Health.Fhir.Core/Features/Search/LightweightReferenceToElementResolver.cs

```diff
@@ -0,0 +1,56 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Serialization;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.Core.Models;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search
+{
+    public class LightweightReferenceToElementResolver : IReferenceToElementResolver
+    {
+        private readonly IReferenceSearchValueParser _referenceParser;
+        private readonly IModelInfoProvider _modelInfoProvider;
+
+        public LightweightReferenceToElementResolver(
+            IReferenceSearchValueParser referenceParser,
+            IModelInfoProvider modelInfoProvider)
+        {
+            EnsureArg.IsNotNull(referenceParser, nameof(referenceParser));
+            EnsureArg.IsNotNull(modelInfoProvider, nameof(modelInfoProvider));
+
+            _referenceParser = referenceParser;
+            _modelInfoProvider = modelInfoProvider;
+        }
+
+        public ITypedElement Resolve(string reference)
+        {
+            if (string.IsNullOrEmpty(reference))
+            {
+                return null;
+            }
+
+            ReferenceSearchValue parsed = _referenceParser.Parse(reference);
+
+            if (parsed == null || !_modelInfoProvider.IsKnownResource(parsed.ResourceType))
```

Comment:
Is null an expected case, or does it signal the possibility of a  problem?

Reply:
Returning null if the element resolution fails seems to be the current behavior. This is happening somewhere mid-fhirpath, so it might indicate a problem, but I'm not sure if throwing is expected because of a malformed reference is encountered